### ISSUE TITLE
[BUG] in sorting/merg_sort.cpp

### ISSUE THREAD

**Issue by ayaankhan98** [TheAlgorithms/C-Plus-Plus#866]
https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/sorting/merge_sort.cpp
Program fails to sort the given array throghs an error 
```
munmap_chunk(): invalid pointer
Aborted (core dumped)
```
## Description

try the program with input
```
Enter the number of elements : 10

Enter the unsorted elements : 
-10 78 -1 -6 7 4 94 5 99 0 
```
## Expected Behavior

Expected to sort the array

## Actual Behavior

throghs an error
```
munmap_chunk(): invalid pointer
Aborted (core dumped)
```

**Comment by kvedala** [comment]
The program is a copy-paste from [here](https://www.geeksforgeeks.org/merge-sort/)
Even the copy-paste done by the author was wrong. Line# 74 should have `size-1` as the input argument. That would resolve the memory access issue

**Comment by ayaankhan98** [comment]
okay no problem let me fix this.

**Comment by ayaankhan98** [comment]
fixed in #872
